Rubberduck.Core/UI/Controls/FindAllReferencesService.cs

```diff
@@ -0,0 +1,228 @@
+        {
+            if (e.State != ParserState.Ready) { return; }
+
+            if (_viewModel == null) { return; }
+
+            _uiDispatcher.InvokeAsync(UpdateTab);
+        }
+
+        public void FindAllReferences(Declaration declaration)
+        {
+            if (_state.Status != ParserState.Ready)
+            {
+                return;
+            }
+
+            var viewModel = CreateViewModel(declaration);
+            if (!viewModel.SearchResults.Any())
+            {
+                _messageBox.NotifyWarn(string.Format(RubberduckUI.AllReferences_NoneFound, declaration.IdentifierName), RubberduckUI.Rubberduck);
+                return;
+            }
+
+            if (viewModel.SearchResults.Count == 1)
+            {
+                _navigateCommand.Execute(viewModel.SearchResults.Single().GetNavigationArgs());
+                return;
+            }
+
+            _viewModel.AddTab(viewModel);
+            _viewModel.SelectedTab = viewModel;
+
+            try
+            {
+                var presenter = _presenterService.Presenter(_viewModel);
+                presenter.Show();
+            }
+            catch (Exception e)
+            {
+                _logger.Error(e);
+            }
+        }
+
+        public void FindAllReferences(Declaration declaration, ReferenceInfo reference)
+        {
+            if (_state.Status != ParserState.Ready ||
+                !(declaration is ProjectDeclaration project))
+            {
+                return;
+            }
+
+            var usages = _state.DeclarationFinder.FindAllReferenceUsesInProject(project, reference, out var referenceProject);
+            if (!usages.Any())
+            {
+                _messageBox.NotifyWarn(string.Format(RubberduckUI.AllReferences_NoneFoundReference, referenceProject.IdentifierName), RubberduckUI.Rubberduck);
+                return;
+            }
+
+            if (usages.Count > 1000 &&
+                !_messageBox.ConfirmYesNo(string.Format(RubberduckUI.AllReferences_PerformanceWarning, referenceProject.IdentifierName, usages.Count),
+                    RubberduckUI.PerformanceWarningCaption))
+            {
+                return;
+            }
+
+            var viewModel = CreateViewModel(project, referenceProject, usages);
+            _viewModel.AddTab(viewModel);
+            _viewModel.SelectedTab = viewModel;
+
+            try
+            {
+                var presenter = _presenterService.Presenter(_viewModel);
+                presenter.Show();
+            }
+            catch (Exception e)
+            {
+                _logger.Error(e);
+            }
+        }
+
+        private string GetModuleLine(QualifiedModuleName module, int line)
+        {
+            var component = _state.ProjectsProvider.Component(module);
+            using (var codeModule = component.CodeModule)
+            {
+                return codeModule.GetLines(line, 1).Trim();
```

Comment:
Would it be a good idea to use `ICodePaneHandler` to get the *logical* code line here?

Reply:
This gets shoved out as the line number in the search results window, and it should be tied directly to an identifier, so it can only be on one line.  Is there a performance benefit or maintenance benefit to using the `ICodePaneHandler`? THB I haven't really looked that closely at it because it would probably nerd snipe me into re-writing the indenter.